client.go

```diff
@@ -360,4 +482,100 @@ func (c *Client) createComponents(config *Config) {
 		Logger:               c.ic.Logger,
 	}
 	c.proxyManager = newProxyManager(proxyManagerServiceBundle)
+	c.db = sql.OpenDB(idriver.NewConnectorWithClient(c.ic, true))
+}
+
+// SQLOptions are server-side query options.
+type SQLOptions struct {
+	cursorBufferSize int32
+	timeout          *int64
+	schema           *string
+	err              error
+}
+
+/*
+SetCursorBufferSize sets the query cursor buffer size.
+When rows are ready to be consumed, they are put into an internal buffer of the cursor.
+This parameter defines the maximum number of rows in that buffer.
+When the threshold is reached, the backpressure mechanism will slow down the execution, possibly to a complete halt, to prevent out-of-memory.
+The default value is expected to work well for most workloads.
+A bigger buffer size may give you a slight performance boost for queries with large result sets at the cost of increased memory consumption.
+Defaults to 4096.
+The given buffer size must be in the non-negative int32 range.
+*/
+func (s *SQLOptions) SetCursorBufferSize(cbs int) {
+	v, err := check.NonNegativeInt32(cbs)
+	if err != nil {
+		s.err = ihzerrors.NewIllegalArgumentError("setting cursor buffer size", err)
+		return
+	}
+	s.cursorBufferSize = v
+}
+
+/*
+SetQueryTimeout sets the query execution timeout.
+If the timeout is reached for a running statement, it will be cancelled forcefully.
+Zero value means no timeout.
+Negative values mean that the value from the server-side config will be used.
+Defaults to -1.
+*/
+func (s *SQLOptions) SetQueryTimeout(t time.Duration) {
+	if t == 0 {
+		s.timeout = nil
+		return
+	}
```

Comment:
Giving 0 to timeout is not the same as -1. If a user gives 0, you set timeout to nil, and then in ExtractTimeoutMillis if timeout is nil, you set it to the default timeout value, which is -1.

Isn't this wrong?

Reply:
That's right. The idea was enabling the user to "unset" a setting, that doesn't work as expected as you pointed out. Fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/728/commits/6d89765f1e4b6dceb68f93e76191d3facb572ae9